source/common/network/utility.cc

```diff
@@ -356,6 +356,17 @@ Address::InstanceConstSharedPtr Utility::getOriginalDst(Socket& sock) {
   }
 
   if (status != 0) {
+#ifdef IP_TRANSPARENT
+    if (Api::OsSysCallsSingleton::get().supportsIpTransparent()) {
+      socklen_t flag_len = sizeof(int);
+      int is_transparent;
+      status =
+          sock.getSocketOption(SOL_IP, IP_TRANSPARENT, &is_transparent, &flag_len).return_value_;
+      if (status == 0 && is_transparent) {
+        return sock.ioHandle().localAddress();
+      }
```

Comment:
would you able to add test for the logic you added?

Reply:
I'd like to but I'm not sure this logic could be unit tested. 
I checked  https://github.com/envoyproxy/envoy/blob/6b83db42c418abca6d21160cf6b1c439657d6ef0/test/common/network/utility_test.cc#L225
and have no idea what is this for ...